src/include/dns.h

```diff
@@ -0,0 +1,349 @@
+	/** Start of Authority record */
+	DNS_TYPE_SOA= 6,
+	/** mailbox domain name record */
+	DNS_TYPE_MB,
+	/** mail group member record */
+	DNS_TYPE_MG,
+	/** mail rename domain name record */
+	DNS_TYPE_MR,
+	/** NULL record */
+	DNS_TYPE_NULL_R,
+	/** well known service description record */
+	DNS_TYPE_WKS,
+	/** Pointer record */
+	DNS_TYPE_PTR= 12,
+	/** Host information record */
+	DNS_TYPE_HINFO, 
+	/** mailbox or mail list information record */
+	DNS_TYPE_MINFO,
+	/** Mail exchanger record */
+	DNS_TYPE_MX= 15,
+	/** Text record */
+	DNS_TYPE_TXT= 16,
+	/** Responsible person record */
+	DNS_TYPE_RP,
+	/** AFS database record */
+	DNS_TYPE_AFSDB,
+	/** DNS X25 resource record */
+	DNS_TYPE_X25,
+	/** Integrated Services Digital Network record */
+	DNS_TYPE_ISDN,
+	/** Route Through record */
+	DNS_TYPE_RT,
+	/** network service access point address record */
+	DNS_TYPE_NSAP,
+	/** network service access point address pointer record */
+	DNS_TYPE_NSAP_PTR,
+	/** Signature record */
+	DNS_TYPE_SIG,
+	/** Key record */
+	DNS_TYPE_KEY,
+	/** Mail Mapping Information record */
+	DNS_TYPE_PX,
+	/** DNS Geographical Position record */
+	DNS_TYPE_GPOS,
+	/** IPv6 address record */
+	DNS_TYPE_AAAA= 28,
+	/**	Location record */
+	DNS_TYPE_LOC,
+	/** Obsolete record */
+	DNS_TYPE_NXT,
+	/** DNS Endpoint Identifier record */
+	DNS_TYPE_EID,
+	/** DNS Nimrod Locator record */
+	DNS_TYPE_NIMLOC,
+	/** Service locator record */
+	DNS_TYPE_SRVS= 33, 
+	/** Asynchronous Transfer Mode address record */
+	DNS_TYPE_ATMA,
+	/** Naming Authority Pointer record */
+	DNS_TYPE_NAPTR= 35,
+	/** Key eXchanger record */
+	DNS_TYPE_KX,
+	/** Certificate record */
+	DNS_TYPE_CERT,
+	/** Obsolete, replaced by AAAA type */
+	DNS_TYPE_A6,
+	/** Delegation Name record */
+	DNS_TYPE_DNAM,
+	/** Kitchen sink record */
+	DNS_TYPE_SINK,
+	/** Option record */
+	DNS_TYPE_OPT,
+	/** Address Prefix List record */
+	DNS_TYPE_APL,
+	/** Delegation signer record */
+	DNS_TYPE_DS,
+	/** SSH Public Key Fingerprint record */
+	DNS_TYPE_SSHFP,
+	/** IPsec Key record */
+	DNS_TYPE_IPSECKEY,
+	/** DNSSEC signature record */
+	DNS_TYPE_RRSIG,
+	/** Next-Secure record */
+	DNS_TYPE_NSEC,
+	/** DNS Key record */
+	DNS_TYPE_DNSKEY,
+	/** DHCP identifier record */
+	DNS_TYPE_DHCID,
+	/** NSEC record version 3 */
+	DNS_TYPE_NSEC3,
+	/** NSEC3 parameters */
+	DNS_TYPE_NSEC3PARAM=51,
+	
+	/** IXFR parameters */
+	DNS_TYPE_IXFR=251,	
+	/** AXFR parameters */
+	DNS_TYPE_AXFR=252,		
+	/** request for Mail Box parameters */
+	DNS_TYPE_MAILB=253,		
+	/** request for Mail Agent resources */
+	DNS_TYPE_MAILA=254,			
+	/** All cached records */
+	DNS_TYPE_ALL = 255
+	
+	/* ...32769 ... 65534  */
+	
+	// 49211
+};
+
+
+/**
+ * An enum for all possible DNS classes
+ */
+enum DnsClass
+{
+	/** Internet class */
+	DNS_CLASS_IN = 1,
+	/** CSNET class */
+	DNS_CLASS_CS = 2,
+	/** Chaos class */
+	DNS_CLASS_CH = 3,
+	/** Hesiod class */
+	DNS_CLASS_HS = 4,
+	/** NONE class */
+	DNS_CLASS_NONE = 0xFE,
```

Comment:
Mixing decimal and hexadecimal values for enum constants does decrease readability.